File: examples/typescript/events/event.ts
import { hatchet } from '../hatchet-client';
import { Input } from './workflow';

async function main() {
  // > Pushing an Event
  const res = await hatchet.events.push<Input>('simple-event:create', {
    Message: 'hello',
    ShouldSkip: false,
  });

  // > Push an Event with Metadata
  const withMetadata = await hatchet.events.push(
    'user:create',
    {
      test: 'test',
    },
    {
      additionalMetadata: {
        source: 'api', // Arbitrary key-value pair
      },
    }
  );

  // > Bulk push events
  const events = [
    {
      payload: { test: 'test1' },
      additionalMetadata: { user_id: 'user1', source: 'test' },
    },
    {
      payload: { test: 'test2' },
      additionalMetadata: { user_id: 'user2', source: 'test' },
    },
    {
      payload: { test: 'test3' },
      additionalMetadata: { user_id: 'user3', source: 'test' },
    },
  ];

  await hatchet.events.bulkPush('user:create', events);

  console.log(res.eventId);
}

if (require.main === module) {
  main();
}

File: examples/ruby/events/event.rb
# frozen_string_literal: true

require "hatchet-sdk"

HATCHET = Hatchet::Client.new

# > Event trigger
HATCHET.event.push("user:create", { "should_skip" => false })

# > Event trigger with metadata
HATCHET.event.push(
  "user:create",
  { "userId" => "1234", "should_skip" => false },
  additional_metadata: { "source" => "api" }
)
